tidy3d/plugins/mode/solver.py

```diff
@@ -1011,6 +1012,26 @@ def split_curl_field_postprocess_inverse(split_curl, E):
         """E has the shape (3, N, num_modes)"""
         raise RuntimeError("Split curl not yet implemented for relative mode solver.")
 
+    @staticmethod
+    def mode_plane_contain_good_metal(material_response) -> bool:
+        """Find out if epsilon on the modal plane contain very metallic materials whose permittivity
+        or permeability value is very large.
+        """
+        if material_response is None:
+            return False
+        return np.any(np.abs(material_response) > 0.9 * np.abs(pec_val))
```

Comment:
I think this 0.9 should still be either a function argument as previously in `any_pec`, or a `GOOD_CONDUCTOR_THRESHOLD` constant defined at the top of the file. Also: I wonder if we want it to be lower than 0.9?

Reply:
Any examples that you have in mind to have it lower than 0.9?

Reply:
Not really. It's more that our `pec_val` is somewhat arbitrary? Like should it be good conductor already at like 1000?

Reply:
In principle, I think good conductor here means it can be treated as impenetrable inside the conductor. So in time-stepping, the fields are set to 0; and in mode solver, sufficiently large permittivity. In our implementation, it's limited to materials that are defined with td.PEC or td.LossyMetalMedium. Otherwise, we cannot just arbitrarily boost its permittivity values.

Reply:
Ok, sounds good. So the threshold of 0.9 is just to make sure we capture PEC and everything that's effectively the same, up to some small tolerance.

Reply:
But as mentioned in the other PR, I'll revert all those and just use simulation.mediums to select precision.